src/solace_ai_connector/components/general/llm/langchain/langchain_chat_model_base.py

```diff
@@ -109,9 +134,18 @@ def invoke(self, message, data):
 
         session_id = data.get("session_id", None)
         clear_history = data.get("clear_history", False)
+        stream = data.get("stream")
+
+        should_stream = self.llm_mode == "stream"
+        if stream is not None:
+            should_stream = stream
```

Comment:
This logic is confusing because when the llm_mode has been configured `stream` statically, we expect streaming regardless of the payload; however, this code indicates that the `stream` value in the payload is higher prior that the llm_mode. I suggest that you add a new llm_mode like `dynamic` which can support both streaming and non-streaming.
In worse case, please ensure that the stream variable is boolean.

Reply:
But does dynamic mean `stream` or `none` by default? 
I want the payload `stream` attribute to be optional, and overwrite only if provided

Reply:
Should I instead add another config like `allow_overwrite_llm_mode` ?

Reply:
Updated